web/packages/teleport/src/Navigation/SideNavigation/Navigation.tsx

```diff
@@ -353,12 +343,12 @@ export function Navigation() {
   );
 
   useEffect(() => {
-    if (stickyMode) {
-      if (currentPageSection) {
-        handleSetExpandedSection(currentPageSection);
-      } else {
-        resetExpandedSection();
-      }
+    // Whenever the user changes page, if stickyMode is enabled and the page is part of the sidenav, the drawer should be expanded with the current page section.
+    // If the page is not part of the sidenav, such as Account Settings, curentPageSection will be undefined.
+    if (stickyMode && currentPageSection) {
```

Comment:
It works differently than before, in a non-sticky mode it will call `collapseDrawer()` whenever `currentPageSection` changes. Won't it cause some unwanted side effects? 

In my previous comments I asked about limiting the number of conditions, but I assume it's not easy to do. At least we can decrease the indentation (same for `collapseDrawer`):
```tsx
  useEffect(() => {
    // Whenever the user changes page, if stickyMode is enabled and the page is part of the sidenav, the drawer should be expanded with the current page section.
    // If the page is not part of the sidenav, such as Account Settings, curentPageSection will be undefined.
    if (!stickyMode) {
      return;
    }

    if (currentPageSection) {
      handleSetExpandedSection(currentPageSection);
    } else {
      collapseDrawer(false);
    }
  }, [currentPageSection]);
```